util/reggen/countermeasure.py

```diff
@@ -0,0 +1,71 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Dict, Sequence
+
+from .lib import check_keys, check_str, check_list
+
+# TO BE DEFINED
+CM_ASSETS = [
+    'FOO'
+]
+
+# TO BE DEFINED
+CM_TYPES = [
+    'BAR'
+]
+
+
+class CounterMeasure:
+    def __init__(self, name: str, desc: str):
+        self.name = name
+        self.desc = desc
+
+    @staticmethod
+    def from_raw(what: str, raw: object, bname: str) -> 'CounterMeasure':
+        rd = check_keys(raw, what, ['name', 'desc'], [])
+
+        name = check_str(rd['name'], 'name field of ' + what)
+        desc = check_str(rd['desc'], 'desc field of ' + what)
+
+        name_split = name.split('.')
+
+        if len(name_split) != 3:
+            raise ValueError('The countermeasure ID {} ({}) has the '
+                             'wrong format.'
+                             .format(name, what))
+
+        if bname.upper() != name_split[0]:
+            raise ValueError('The block name of countermeasure {} ({})'
+                             'is {} but should be {}.'
+                             .format(name, what, name_split[0], bname))
```

Comment:
I'd suggest making `CounterMeasure` oblivious to the enclosing block and moving this check to `ip_block.py`. So `CounterMeasure.from_raw` is just parsing a string. The "does this make sense in this block" check should probably live in the block.

Reply:
Ok, let me move these checks.

Reply:
Actually, the `bname` check is not needed anymore once we drop the block name from the countermeasure naming scheme. As @sriyerg pointed out, the block name can be inferred from the `IpBlock` name.

Reply:
Even better!